src/algebraic_geometry/EllipticCurve/point.lean

```diff
@@ -0,0 +1,328 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import algebraic_geometry.EllipticCurve
+import field_theory.galois -- temporary import to enable point notation
+import tactic.field_simp
+
+/-!
+# The group of rational points on an elliptic curve over a field
+
+Let $E$ be an elliptic curve over a field $F$. For a field extension $K$ of $F$, a $K$-rational
+point on $E$ is simply a point $[X:Y:Z] \in \mathbb{P}^2$ over $K$ in the projective plane cubic
+defined by $Y^2Z + a_1XYZ + a_3YZ^2 = X^3 + a_2X^2Z + a_4XZ^2 + a_6Z^3$. This file defines the type
+of $K$-rational points on $E$ as an inductive, since any such point either lies in the affine chart
+$Z \ne 0$ and satisfies the Weierstrass equation obtained by setting $x = X/Z$ and $y = Y/Z$, or is
+the unique point at infinity $\mathcal{O} = [0:1:0] \in \mathbb{P}^2$ obtained by setting $Z = 0$.
+
+The set of $K$-rational points on $E$ forms an abelian group under a chord-and-tangent process.
+ * The identity point is $\mathcal{O}$.
+ * Given a point $P$, its negation $-P$ is defined to be the unique third point of intersection
+    between $E$ and the line joining $\mathcal{O}$ and $P$, which exists by Bézout's theorem.
+    Explicitly, if $P = (x, y)$, then $-P = (x, -y - a_1x - a_3)$.
+ * Given two points $P$ and $Q$, their addition $P + Q$ is defined to be the negation of the unique
+    third point of intersection between $E$ and the line joining $P$ and $Q$, which again exists by
+    Bézout's theorem. Explicitly, let $P = (x_1, y_1)$ and $Q = (x_2, y_2)$. If $x_1 = x_2$ and
+    $P \ne -Q$, then this line is the tangent of $E$ at $P = Q$ and has a well-defined gradient
+    $\ell = (3x_1^2 + 2a_2x_1 + a_4 - a_1y_1) / (2y_1 + a_1x_1 + a_3)$. Otherwise $P \ne Q$ and this
+    line has a well-defined gradient $\ell = (y_1 - y_2) / (x_1 - x_2)$. The $x$-coordinate of
```

Comment:
I was thinking that y_1 = -y_2 - a_1x_1 - a_3 (i.e. P = -Q) would be part of my statement at the end that "the group law on this set is then uniquely determined by these constructions", since -Q + Q = 0 in any group so this is automatic.

Reply:
The point I wanted to make is that I read "Otherwise" as "if it is not the case that x_1 = x_2 and P \ne -Q then...", which allows the case P = -Q (if P \ne -P), but then what follows does not really make sense, since \ell is not defined (or at least not in a way that allows to solve for x). So it's just about the documentation, which (I think) should give a complete specification of addition. I'd suggest to start with "if P = -Q, then P + Q = \mathcal{O}. Otherwise ..."

Reply:
Sounds good.